### PR TITLE
Vet clinic database: create animals table

### PR DESCRIPTION
**Pull request by oovillagran**
<h1>Hi there, 👋</h1>

In this branch, I used a relational database to create the initial data structure for a vet clinic. I created a table to store animals' information, insert some data into it, and query it.

Here are the *Project Requirements*:

- Use [this template](https://github.com/microverseinc/curriculum-template-databases) to generate your repo.
- Create a database named vet_clinic.
- Create a table named animals with the following columns:
     - id: integer
     - name: string
     - date_of_birth: date
     - escape_attempts: integer
     - neutered: boolean
     - weight_kg: decimal
- Copy the SQL you used in the previous point into the schema.sql file in the generated repository.
- Insert the following data:
     - Animal: His name is Agumon. He was born on Feb 3rd, 2020, and currently weighs 10.23kg. He was neutered and he has never tried to escape.
     - Animal: Her name is Gabumon. She was born on Nov 15th, 2018, and currently weighs 8kg. She is neutered and she has tried to escape 2 times.
     - Animal: His name is Pikachu. He was born on Jan 7th, 2021, and currently weighs 15.04kg. He was not neutered and he has tried to escape once.
     - Animal: Her name is Devimon. She was born on May 12th, 2017, and currently weighs 11kg. She is neutered and she has tried to escape 5 times.
- Copy the SQL you used in the previous point into the data.sql file in the generated repository.

*Write queries for the following quests:*
- Find all animals whose name ends in "mon".
- List the name of all animals born between 2016 and 2019.
- List the name of all animals that are neutered and have less than 3 escape attempts.
- List the date of birth of all animals named either "Agumon" or "Pikachu".
- List name and escape attempts of animals that weigh more than 10.5kg
.
- Find all animals that are neutered.
- Find all animals not named Gabumon.
- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
.

**Take a screenshot of the results of your queries**
- Copy the SQL you used in the previous point into the queries.sql file in the generated repository.
- Remember to add the screenshot with the results of your queries to your Pull Request description.
screenshot vet clinic 1
screenshot vet clinic 2

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,5 +1,10 @@
 /* Database schema to keep the structure of entire database. */
 
 CREATE TABLE animals (
-    name varchar(100)
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  name VARCHAR NOT NULL,
+  date_of_birth DATE NOT NULL,
+  escape_attempts INT NOT NULL,
+  neutered bit NOT NULL,
```
Comment (Ismailco) [inline schema.sql]: - [x] Nice job creating the animals table however the type of `neutered` should be **boolean**, kindly make sure to fix this column

File: queries.sql [hunk 2]
```diff
@@ -1,3 +1,51 @@
...
+SELECT *
+FROM animals
+WHERE name LIKE '%mon';
+
+
+-- List the name of all animals born between 2016 and 2019.
+
+SELECT *
+FROM animals
+WHERE date_of_birth > '2016-01-01' and date_of_birth < '2019-12-31';
```
Comment (Ismailco) [inline queries.sql]: - [x] In this query you should only select names of animals with data of birth between 2016 and 2019, kindly make sure to return only the names, not all the table data

File: queries.sql [hunk 3]
```diff
@@ -1,3 +1,51 @@
...
+
+-- List the name of all animals that are neutered and have less than 3 escape attempts.
+
+SELECT *
+FROM animals
+WHERE neutered = true and escape_attempts < 3;
+
+-- List the date of birth of all animals named either "Agumon" or "Pikachu".
```
Comment (Ismailco) [inline queries.sql]: - [x] Same here you're returning all the table data. Kindly make sure to return only the names of all animals that are neutered and have less than 3 escape attempts

File: queries.sql [hunk 4]
```diff
@@ -1,3 +1,51 @@
...
+
+SELECT name, date_of_birth
+FROM animals
+WHERE name = 'Agumon' or name ='Pikachu';
```
Comment (Ismailco) [inline queries.sql]: - [x] Kindly make sure to select only the date_of_birth here, the names are not required

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @oovillagran, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
**- [ ] Kindly consider updating the screenshots after making the requested changes**
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (oovillagran) [reply schema.sql]: Done!

File: queries.sql (hunk 2 above)
Comment (oovillagran) [reply queries.sql]: Done!

File: queries.sql (hunk 3 above)
Comment (oovillagran) [reply queries.sql]: Done!

File: queries.sql (hunk 4 above)
Comment (oovillagran) [reply queries.sql]: Done!

**Comment by oovillagran** [comment]
Thanks @Ismailco !

**Comment by oovillagran** [comment]
<h1>Hi there, 👋</h1>

Here is the updated screenshot of the SQL vet_clinic database.

screenshot vet clinic

**Review by danmainah — APPROVED** [review]
Hi @oovillagran , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  All queries added correctly and screenshots updated as requested

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oovillagran** [comment]
Thanks @danmainah !👍🚀🎉

**Pull request merged**